app/src/main/java/in/co/rajkumaar/amritarepo/timetable/AcademicTimetableActivity.java

```diff
@@ -1,303 +1,347 @@
+        } else {
+            courses = new ArrayList<>();
+            Gson gson = new Gson();
+            String json = pref.getString("courses", null);
+            Type listType = new TypeToken<ArrayList<String>>() {
+            }.getType();
+            ArrayList<String> timings = gson.fromJson(json, listType);
+            courses.addAll(timings);
+        }
+        ArrayAdapter<String> courseAdapter = new ArrayAdapter<>(AcademicTimetableActivity.this, R.layout.spinner_item1, courses);
+        courseAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        course.setAdapter(courseAdapter);
+
+        sems = new ArrayList<>();
+        sems.add("[Choose semester]");
+        int i = 1;
+        while (i <= 10) {
+            sems.add(String.valueOf(i++));
+        }
+        ArrayAdapter<String> semAdapter = new ArrayAdapter<>(this, R.layout.spinner_item1, sems);
+        semAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        sem.setAdapter(semAdapter);
+
+        batches.add("[Choose batch]");
+        batches.add("A");
+        batches.add("B");
+        batches.add("C");
+        batches.add("D");
+        batches.add("E");
+        batches.add("F");
+        ArrayAdapter<String> batchAdapter = new ArrayAdapter<>(this, R.layout.spinner_item1, batches);
+        batchAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        batch.setAdapter(batchAdapter);
+
+        ArrayAdapter<String> yearAdapter = new ArrayAdapter<>(this, R.layout.spinner_item1, Utils.getAcademicYears());
+        yearAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        year.setAdapter(yearAdapter);
+
+    }
+
+    private HashMap<String, String> initializehashmap(HashMap<String, String> parse) {
+        parse.put("Int MSc", "IMSc");
+        parse.put("MSc", "MSC");
+        parse.put("Int MA", "IMA");
+        return parse;
+    }
+
+    private void getCourse() {
+        HashMap<String, String> parse = new HashMap<>();
+        parse = initializehashmap(parse);
+        AsyncHttpClient client = new AsyncHttpClient();
+        final HashMap<String, String> finalParse = parse;
+        client.get("https://intranet.cb.amrita.edu/TimeTable/", new AsyncHttpResponseHandler() {
+            @Override
+            public void onSuccess(int i, Header[] headers, byte[] bytes) {
+                Document html = Jsoup.parse(new String(bytes));
+                Elements rows = html.select("center > table").select("tbody>tr").select("td>table").select("tbody>tr").select("td>form").select("select");
+                for (Element row : rows) {
+                    Elements columns = row.getElementsByTag("option");
+                    if (columns.get(0).text().equals("Course")) {
+                        for (int j = 1; j < columns.size(); j++) {
+                            String parsedCourse = columns.get(j).text().replace(".", "");
```

Comment:
Try to avoid this when you fetch the HashMap also from the `select` tag : 
![image](https://user-images.githubusercontent.com/37476886/77320079-ecb1f000-6d35-11ea-8c93-c2d3dbcb68d0.png)